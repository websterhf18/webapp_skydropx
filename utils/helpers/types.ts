export type rateData = {
  id: string;
  attributes: {
    days: number;
    amount_local: string;
    provider: string;
    currency_local: string;
  };
};
export type shippingData = {
  id: string;
  attributes: {
    status: string;
  };
  relationships: {
    address_from: any;
    address_to: any;
    parcels: any;
  };
};
