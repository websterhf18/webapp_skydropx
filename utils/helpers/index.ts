import _ from "lodash";
import { ShippingFormValues } from "../../components/Forms/Shipping.form";
import {
  AddressFrom,
  AddressTo,
  createShipmentsResponse,
  Parcel,
} from "../api/skydropx.types";
import { BASE_REQUEST } from "../constants";

export const combineFormWithBase = (valuesForm: ShippingFormValues) => {
  const { address_from, address_to } = BASE_REQUEST;
  const newOrigen: AddressFrom = {
    ...address_from,
    zip: valuesForm.codigo_origen,
  };
  const newDestino: AddressTo = {
    ...address_to,
    zip: valuesForm.codigo_destino,
  };
  const newMedidas: Array<Parcel> = [
    {
      weight: valuesForm.peso || 0,
      distance_unit: "CM",
      mass_unit: "KG",
      height: valuesForm.altura || 0,
      width: valuesForm.ancho || 0,
      length: valuesForm.largo || 0,
    },
  ];
  return {
    ...BASE_REQUEST,
    address_from: newOrigen,
    address_to: newDestino,
    parcels: newMedidas,
  };
};

export const cleanResponseRates = (response: createShipmentsResponse) => {
  const { included } = response;
  const newRates = _.filter(included, (include) => {
    return include.type === "rates";
  });
  const orderRates = _.orderBy(
    newRates,
    ["attributes.days", "attributes.amount_local"],
    ["asc", "asc"],
  );
  return orderRates;
};
export const findByIdAddress = (id: string, arrayIncludes: Array<any>) => {
  const objectInclude = _.filter(arrayIncludes, (include) => {
    return include.id === id;
  });
  const [objectFirst] = objectInclude;
  return objectFirst;
};
