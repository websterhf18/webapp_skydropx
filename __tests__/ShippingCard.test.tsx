/**
 * @format
 * @jest-environment jsdom
 */
import React from "react";
import renderer from "react-test-renderer"; // Note: test renderer must be required after react-native.
import ShippingRateCard from "../components/HomePage/ShippingRate.card";
import { rateData } from "../utils/helpers/types";

import { Provider } from "react-redux";
import store from "../models/index";

const rateDataTest: rateData = {
  id: "000",
  attributes: {
    days: 0,
    amount_local: "139",
    provider: "UPS",
    currency_local: "MXN",
  },
};

it("Renders the Shipping Rate Card Component", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ShippingRateCard indexCard={0} rateData={rateDataTest} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
