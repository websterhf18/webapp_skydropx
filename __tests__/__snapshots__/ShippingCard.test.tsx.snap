// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Shipping Rate Card Component 1`] = `
<div
  className="card  m-4 p-2 shadow promo-card"
  data-label="Mejor Opcion"
>
  <div
    className="card-body"
  >
    <div
      className="row text-center"
    >
      <div
        className="col-md-3"
      >
        <h5
          className="card-title"
        >
          Proveedor
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted"
        >
          UPS
        </h6>
      </div>
      <div
        className="col-md-3"
      >
        <h5
          className="card-title"
        >
          Tiempo entrega
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted"
        >
          0
        </h6>
      </div>
      <div
        className="col-md-3"
      >
        <h5
          className="card-title"
        >
          Precio
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted"
        >
          $ 
          139
           
          MXN
        </h6>
      </div>
      <div
        className="col-md-3"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Crear etiqueta
        </button>
      </div>
    </div>
  </div>
</div>
`;
